Sources_v3/Database/DatabaseSession_Tests.swift

```diff
@@ -118,4 +118,39 @@ class DatabaseSession_Tests: StressTestCase {
         let message = try XCTUnwrap(loadedMessage)
         XCTAssert(loadedChannel.latestMessages.contains(message))
     }
+    
+    func test_saveEvent_unreadCountFromEventPayloadIsApplied() throws {
+        let eventPayload = EventPayload<DefaultDataTypes>(eventType: .messageNew,
+                                                          connectionId: .unique,
+                                                          cid: .unique,
+                                                          currentUser: .dummy(userId: .unique,
+                                                                              role: .user,
+                                                                              unreadCount: nil),
+                                                          unreadCount: .dummy)
+
+        _ = try await {
+            database.write({ session in
+                try! session.saveEvent(payload: eventPayload)
+            }, completion: $0)
+        }
```

Comment:
☝️

Reply:
fixed in 3aa53f22af7648b02df547f0a7e0bf9aaca0af30